src/RestController.php

```diff
@@ -998,4 +973,46 @@ protected function createHalEntity($entity)
             $this->getRouteIdentifierName()
         );
     }
+
+    /**
+     * @param Exception $exception
+     * @return ApiProblem
+     */
+    private function createApiProblemFromException(Exception $exception)
+    {
+        return new ApiProblem(
+            $this->validateHttpStatusCode($exception->getCode()),
+            $exception->getMessage()
+        );
+    }
+
+    /**
+     * @param Throwable $throwable
+     * @return ApiProblem
+     */
+    private function createApiProblemFromThrowable(Throwable $throwable)
+    {
+        return new ApiProblem(
+            $this->validateHttpStatusCode($throwable->getCode()),
+            $throwable->getMessage()
+        );
+    }
+
+    /**
+     * Ensure we have a valid HTTP status code for an ApiProblem
+     *
+     * @param int|string $code
+     * @return int
+     */
+    private function validateHttpStatusCode($code)
```

Comment:
If the `createApiProblemFrom*()` methods are merged, technically you wouldn't need to separate this into a new method. However, I think it makes the other methods more readable, so I'm fine if this stays as submitted.